README.md

```diff
@@ -14,30 +28,99 @@ To install multiple versions of Node.js, you may find it easier to use a node ve
 * [nvm](https://github.com/creationix/nvm)
 * [n](https://github.com/tj/n)
 
-### Installation
+
+## Installation
+
+Clone this repo and install its dependencies...
 
 ```
 $ npm install
 ```
 
-### Usage
+You'll then need to duplicate the `config.sample.json` file as `config.json` and update its contents...
 
 ```
-$ node index.js <searchString>
+{
+  "api": {
+    "protocol": "",
+    "host": "",
+    "paths": [],
+    "headers": {}
+  },
+  "whitelist": []
+}
 ```
 
-### Tests
+* **api.protocol**: The protocol that should be used to communicate with the API (**Required**)
+* **api.host**: The domain name of the API to fetch a list of repos from (**Required**)
+* **api.paths**: The endpoint(s) of the API to fetch a list of repos from (**Required**)
+* **api.headers**: Any request headers that the API requires (such as auth tokens) (**Optional**)
+
+
+## Usage
+
+```
+$ node index.js searchString
+```
+Where `searchString`...
+
+* Is the exact string to search for
+* Should not include and punctuation at the beginning of selectors
+	* `.className` should be `className`
+	* `#idName` should be `idName`
+
+#### Results
+
+Results appear in the console like this...
+
+```
+repository-name (enterprise|public) [n]
+```
+
+* **repository-name**: The name of the repository
+* **(enterprise|public)**: Whether the repository lives in the open or in an enterprise instance of github
+* **[n]**: The number of occurrences of `searchString` in `repository-name`
+
+They are also saved to a `results.json` file like this...
```

Comment:
no need for `...` here, a colon `...file like:` I think is fantastically beautiful!